setup.py

```diff
@@ -352,22 +351,21 @@ def _no_device() -> bool:
 
 
 def _is_cuda() -> bool:
-    has_cuda = torch.version.cuda is not None
-    return (VLLM_TARGET_DEVICE == "cuda" and has_cuda
-            and not (_is_neuron() or _is_tpu() or _is_hpu()))
+    return VLLM_TARGET_DEVICE == "cuda" and not (_is_neuron() or _is_tpu()
+                                                 or _is_hpu())
 
 
 def _is_hip() -> bool:
-    return (VLLM_TARGET_DEVICE == "cuda"
-            or VLLM_TARGET_DEVICE == "rocm") and torch.version.hip is not None
+    return VLLM_TARGET_DEVICE == "rocm"
```

Comment:
I want to stress out this change: I think it's better to explicitly set the ROCm target device rather than inferring it based on the value of `torch.version.hip`. `Dockerfile.rocm` has been updated accordingly.

Reply:
why do you think the exception is good?

Reply:
Can you elaborate?